src/Vts/MonteCarlo/Detectors/TransmittedMTOfXAndYAndSubregionHistDetector.cs

```diff
@@ -264,104 +265,32 @@ public void Normalize(long numPhotons)
-                                    for (var l = 0; l < NumSubregions; l++) {
-                                        for (var m = 0; m < FractionalMTBins.Count + 1; m++)
-                                        {
-                                            FractionalMT[i, j, k, l, m] = binaryReader.ReadDouble();
-                                        }
-                                    }
-                                }                              
-                            }
-                        }
-                    }
-                },
-                // return a null serializer, if we're not serializing the second moment
-                !TallySecondMoment ? null :  new BinaryArraySerializer {
-                    DataArray = SecondMoment,
-                    Name = "SecondMoment",
-                    FileTag = "_2",
-                    WriteData = binaryWriter => {
-                        if (!TallySecondMoment || SecondMoment == null) return;
-                        for (var i = 0; i < X.Count - 1; i++) {
-                            for (var j = 0; j < Y.Count - 1; j++) {
-                                for (var k = 0; k < MTBins.Count - 1; k++)
-                                {
-                                    binaryWriter.Write(SecondMoment[i, j, k]);
-                                }
-                            }
-                        }
-                    },
-                    ReadData = binaryReader => {
-                        if (!TallySecondMoment || SecondMoment == null) return;
-                        SecondMoment = new double[ X.Count - 1, Y.Count, MTBins.Count - 1];
-                        for (var i = 0; i < X.Count - 1; i++) {
-                            for (var j = 0; j < Y.Count - 1; j++) {
-                                for (var k = 0; k < MTBins.Count - 1; k++)
-                                {
-                                    SecondMoment[i, j, k] = binaryReader.ReadDouble();
-                                }
-                            }
-                        }
-                    },
-                },
+            Mean ??= new double[X.Count - 1, Y.Count - 1, MTBins.Count - 1];
+            FractionalMT ??= new double[X.Count - 1, Y.Count - 1, MTBins.Count - 1, NumSubregions, FractionalMTBins.Count + 1];
+            if (TallySecondMoment)
+            {
+                SecondMoment ??= new double[X.Count - 1, Y.Count - 1, MTBins.Count - 1];
+            }
+            var allSerializers = new List<BinaryArraySerializer>
+            {
+                BinaryArraySerializerFactory.GetSerializer(
+                    Mean, "Mean", ""),
+                BinaryArraySerializerFactory.GetSerializer(
+                    FractionalMT, "FractionalMT", "_FractionalMT"),
+                TallySecondMoment ? BinaryArraySerializerFactory.GetSerializer(
+                    SecondMoment, "SecondMoment", "_2") : null,
+

```

Comment:
Extra line before the closing bracket.

Reply:
All review comments implemented.  Thank you for all of your help!